### PR TITLE
Bump versions and dependencies for 1.0 pre-release

### PR DESCRIPTION
**Pull request by nategraf**
Bump versions and dependencies for 1.0 pre-release, following `RELEASE.md`.

### CODE REVIEW

**Review by capossele — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -3,23 +3,23 @@ resolver = "2"
...
 
 [workspace.dependencies]
 # Intra-workspace dependencies
-risc0-build-ethereum = { version = "0.11.0-alpha.1", default-features = false, path = "build" }
-risc0-ethereum-contracts = { version = "0.11.0-alpha.1", default-features = false, path = "contracts" }
-risc0-steel = { version = "0.11.0-alpha.1", default-features = false, path = "steel" }
-risc0-forge-ffi = { version = "0.11.0-alpha.1", default-features = false, path = "ffi" }
+risc0-build-ethereum = { version = "1.0.0-rc.6", default-features = false, path = "build" }
+risc0-ethereum-contracts = { version = "1.0.0-rc.6", default-features = false, path = "contracts" }
+risc0-steel = { version = "1.0.0-rc.6", default-features = false, path = "steel" }
```
Comment (nategraf) [inline Cargo.toml:16]: Reduce this version

File: RELEASE.md
Comment (nategraf) [inline RELEASE.md:1]: Add a note about cherry-picking for patch releases

**Comment by nategraf** [comment]
I'm also going to rebase the `release/1.0` branch and this PR to the latest commit from `main`

File: Cargo.toml (hunk 1 above)
Comment (nategraf) [reply Cargo.toml]: Addressed in 979f5fcd9abe4ccceb163ffb1c76d0273de64bf9

**Comment by nategraf** [comment]
Moved updates to `RELEASE.md` to https://github.com/risc0/risc0-ethereum/pull/130

**Comment by nategraf** [comment]
I'm merging this PR now, such that the `release/1.0` will reflect the latest pre-release.

**Pull request merged**